components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.html

```diff
@@ -0,0 +1,110 @@
+                  Display Name is required.
+                </chef-error>
+                <chef-error *ngIf="conflictError">
+                  Environment "{{firstFormGroup.get('name').value}}" already exists.
+                </chef-error>
+              </chef-form-field>
+            </div>
+            <div class="input-margin">
+              <chef-form-field>
+                <label>
+                  <span class="label">Description</span>
+                  <textarea chefInput name="description" id="description" formControlName="description" autofocus></textarea>
+                </label>
+                <chef-error
+                  *ngIf="(firstFormGroup.get('description').hasError('required') || firstFormGroup.get('description').hasError('pattern')) && firstFormGroup.get('description').dirty">
+                  Description is required.
+                </chef-error>
+              </chef-form-field>
+            </div>
+            <div class="button-bar-box">
+              <chef-button tertiary (click)="closeCreateModal()">Cancel</chef-button>
+              <chef-button [disabled]="!firstFormGroup?.valid || conflictError"  primary id="create-button-object-modal" (click)="createEnvironment()">
+                <chef-loading-spinner *ngIf="creating"></chef-loading-spinner>
+                <span *ngIf="!creating">Create</span>
+                <span *ngIf="creating">Creating Environment...</span>
+              </chef-button>
+            </div>
+          </form>
+        </app-infra-tab>
+        <app-infra-tab tabTitle="Constraints" [active]="constraintsTab" [disabled]="!firstFormGroup?.valid">
+          <div *ngIf="showConstraint">
+            <app-infra-environment-constraint
+              [constraintKeys]="constraintKeysp"
+              (constraintValues)="constraintItemsHandler($event)"
+              [dynamicArray]="dynamicArrayp"
+              [name_id]="name_idp">
+
+            </app-infra-environment-constraint>
+          </div>
+          <div class="button-bar-box">
+            <chef-button tertiary (click)="closeCreateModal()">Cancel</chef-button>
+            <chef-button [disabled]="!constraintArray.length || conflictError"  primary id="create-button-object-modal" (click)="createEnvironment()">
+              <chef-loading-spinner *ngIf="creating"></chef-loading-spinner>
+              <span *ngIf="!creating">Create</span>
+              <span *ngIf="creating">Creating Environment...</span>
+            </chef-button>
+          </div>
+        </app-infra-tab>
+        <app-infra-tab tabTitle="Default Attributes" [active]="defaultTab" [disabled]="!firstFormGroup?.valid">
+          <form [formGroup]="thirdFormGroup">
+            <chef-form-field>
+              <textarea chefInput name="dattr" id="dattr" formControlName="dattr" (change)="onChangeJSON($event)"
+                (keyup)="onChangeJSON($event)" [(ngModel)]="default_attr_value">
```

Comment:
One of the main design tenets of Automate is one-way data flow, which means do not use the `[(ngModel)]` construct. Please rework.

Reply:
yes, I have updated the changes.